File: app/src/main/java/me/lx/sample/group/entity/GroupEntity.java
package me.lx.sample.group.entity;

import androidx.databinding.ObservableArrayList;

/**
 * 组数据的实体类
 */
public class GroupEntity {

    private String header;
    private String footer;
    private ObservableArrayList<ChildEntity> childList;

    public GroupEntity(String header, String footer, ObservableArrayList<ChildEntity> childList) {
        this.header = header;
        this.footer = footer;
        this.childList = childList;
    }

    public String getHeader() {
        return header;
    }

    public void setHeader(String header) {
        this.header = header;
    }

    public String getFooter() {
        return footer;
    }

    public void setFooter(String footer) {
        this.footer = footer;
    }

    public ObservableArrayList<ChildEntity> getChildList() {
        return childList;
    }

    public void setChildren(ObservableArrayList<ChildEntity> childList) {
        this.childList = childList;
    }
}

File: app/src/main/java/me/lx/sample/group/entity/GroupEntity.kt
package me.lx.sample.group.entity

import androidx.databinding.ObservableArrayList

/**
 * 组数据的实体类
 */
class GroupEntity {
    var header: String? = null
    var footer: String? = null
    var childList: ObservableArrayList<ChildEntity> = ObservableArrayList()


    constructor(header: String?, footer: String?, childList: ObservableArrayList<ChildEntity>) {
        this.header = header
        this.footer = footer
        this.childList = childList
    }

    fun setChildren(childList: ObservableArrayList<ChildEntity>) {
        this.childList = childList
    }
}